src/cpp/src/speculative_decoding/speculative_decoding_perf_metrics.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+// #pragma once
+
+#include <chrono>
+#include <map>
+#include <string>
+#include <vector>
+#include <ostream>
+#include <iostream>
+
+#include "openvino/genai/perf_metrics.hpp"
+#include "openvino/genai/speculative_decoding/perf_metrics.hpp"
+
+namespace ov::genai {
+
+MeanStdPair calc_mean_and_std(const std::vector<MicroSeconds>& durations);
+
+MeanStdPair ov::genai::SDPerfMetrics::get_ttst() {
+    evaluate_statistics();
+    return ttst;
+};
+
+MeanStdPair ov::genai::SDPerfMetrics::get_latency() {
+    evaluate_statistics();
+    return ttl;
+};
+
+size_t ov::genai::SDPerfMetrics::get_num_accepted_tokens() {
+    evaluate_statistics();
+    return num_accepted_tokens;
+};
+
+size_t ov::genai::SDPerfMetrics::get_num_iterations() {
+    evaluate_statistics();
+    return num_iterations;
+};
+
+void ov::genai::SDPerfMetrics::evaluate_statistics(std::optional<TimePoint> start_time) {
+    if (m_evaluated)
+        return;
+
+    std::vector<ov::genai::MicroSeconds> second_token_duration;
+    std::vector<ov::genai::MicroSeconds> topt_durations;
+    std::vector<ov::genai::MicroSeconds> latency_durations;
+    std::vector<ov::genai::MicroSeconds> apt_durations;
+    if (raw_metrics.m_durations.size() > 0 && raw_metrics.m_batch_sizes.size() > 0) {
+        auto& durations = raw_metrics.m_durations;
+        auto& batch_sizes = raw_metrics.m_batch_sizes;
+
+        raw_metrics.m_times_to_first_token.clear();
+        raw_metrics.m_times_to_first_token.emplace_back(durations[0]);
+
+        num_generated_tokens = batch_sizes[0];
+        num_accepted_tokens = 0;
+
+        raw_metrics.generate_durations.clear();
+        raw_metrics.generate_durations.emplace_back(durations[0]);
+
+        num_iterations = durations.size();
+
+        for (size_t i = 1; i < durations.size(); ++i) {
+            if (i == 1) {
+                // TODO: ttst should be based on time per token or latency?
```

Comment:
I think time per token and latency are the same things here because there's just one token